bundle/edu.gemini.pot/src/main/scala/edu/gemini/spModel/gemini/ghost/GhostAsterism.scala

```diff
@@ -87,242 +83,132 @@ object GhostAsterism {
+      case TargetPlusSky(t, s) => TargetPlusSky(t.copyWithClonedTarget, s)
+      case SkyPlusTarget(s, t) => SkyPlusTarget(s, t.copyWithClonedTarget)
+    }
 
-    implicit val EqualFiberAgitatorState: Equal[FiberAgitatorState] =
-      Equal.equalA[FiberAgitatorState]
+    def defaultBasePosition(when: Option[Instant]): Option[Coordinates] = this match {
+      case SingleTarget(t)    => t.coordinates(when)
+      case DualTarget(t1,t2)  => interpolateCoords(t1.coordinates(when), t2.coordinates(when))
+      case TargetPlusSky(t,s) => interpolateCoords(t.coordinates(when), Some(s))
+      case SkyPlusTarget(s,t) => interpolateCoords(Some(s), t.coordinates(when))
+    }
   }
 
-  /** The high resolution asterism comes in two flavors, normal high resolution
-    * mode and precision radial velocity (PRV) mode.
-    */
-  sealed trait HighResMode extends Product with Serializable
-
-  object HighResMode {
+  object GhostStandardResTargets {
+    def interpolateCoords(c1Opt: Option[Coordinates], c2Opt: Option[Coordinates]): Option[Coordinates] = for {
+      c1 <- c1Opt
+      c2 <- c2Opt
+    } yield c1.interpolate(c2, 0.5)
 
-    /** Normal high res mode can use bright or faint binning options. (Prv is
-      * always bright).
-      */
-    case object Normal extends HighResMode
-
-    /** Prv can be set up with a fiber agitator. The simultaneous calibration
-      * lamp can default to an appropriate value for the target magnitude, or
-      * can be explicitly set to a particular duration.
-      *
-      * (TODO: Do we need to be able to turn off the lamp altogether?  Perhaps
-      * that is the same as setting the duration to 0.)
-      */
-    final case class Prv(agitator: FiberAgitatorState, calDuration: Option[Duration]) extends HighResMode
+    case class SingleTarget(target: GhostTarget) extends GhostStandardResTargets
+    case class DualTarget(target1: GhostTarget, target2: GhostTarget) extends GhostStandardResTargets
+    case class TargetPlusSky(target: GhostTarget, sky: Coordinates) extends GhostStandardResTargets
+    case class SkyPlusTarget(sky: Coordinates, target: GhostTarget) extends GhostStandardResTargets
   }
 
-  /** High resolution GHOST asterism.
+
+  /** High resolution mode.
+    *
+    * It has a calibration duration (of type Duration), which should be determined
+    * by exposure time.
     *
     * The target is always observed using the high resolution IFU1.  The sky
     * coordinates are observed using the sky fibers of IFU2, not SRIFU2. The
     * guide fibers will be used by default because the target must be bright,
     * but can be explicitly turned off.
     */
-  final case class HighRes(
-                     mode:        HighResMode,
-                     ghostTarget: GhostTarget,
-                     sky:         Coordinates) extends GhostAsterism {
+  sealed case class HighResolution(ghostTarget: GhostTarget,
```

Comment:
Did you mean to change this to `sealed`?  I think it should be left as `final` because there should be no subclasses.

Reply:
Ah, I see what I did. I just deleted the code up to `HighResMode` and renamed that. Yes, it should be `final`. I'll make those changes.